civis/ml/_model.py

```diff
@@ -268,6 +257,132 @@ def _show_civisml_warnings(warn_list):
             warnings.warn(warn_str, RuntimeWarning)
 
 
+def _get_job_type_version(alias):
+    """Derive the job type and version from the given alias.
+
+    Parameters
+    ----------
+    alias : str
+        CivisML alias
+
+    Returns
+    -------
+    str
+        Job type, one of {training, prediction, registration}.
+    str
+        CivisML version, e.g., "v2.2".
+    """
+    # A version-less alias for production, e.g., "civis-civisml-training"
+    match_production = re.search(r'\Acivis-civisml-(\w+)\Z', alias)
+    # A versioned alias, e.g., "civis-civisml-training-v2-3"
+    match_v = re.search(r'\Acivis-civisml-(\w+)-v(\d+)-(\d+)\Z', alias)
+    # A special-version alias, e.g., "civis-civisml-training-dev"
+    match_special = re.search(r'\Acivis-civisml-(\w+)-(\S+[^-])\Z', alias)
+
+    if match_production:
+        job_type = match_production.group(1)
+        version = None
+    elif match_v:
+        job_type = match_v.group(1)
+        version = 'v%s.%s' % match_v.group(2, 3)
+    elif match_special:
+        job_type = match_special.group(1)
+        version = match_special.group(2)
+    else:
+        msg = ('Unable to parse the job type and version '
+               'from the CivisML alias "%r"')
+        raise ValueError(msg % alias)
+
+    return job_type, version
+
+
+@lru_cache()
+def _get_template_ids_all_versions(client):
+    """Get templates IDs for all accessible CivisML versions.
+
+    Parameters
+    ----------
+    client : APIClient
+        Civis API client object
+
+    Returns
+    -------
+    Dict[str, Dict[str, int]]
+        Mapping between versions (e.g., "v2.2") and template IDs for the given
+        version (e.g., {'training': 1, 'prediction': 2, 'registration': 3}).
+    """
+    template_alias_objects = client.aliases.list(
+        object_type='template_script', iterator=True
+    )
+    civisml_template_alias_objects = find(
+        template_alias_objects,
+        alias=lambda alias: alias.startswith('civis-civisml-')
+    )
+    ids = collections.defaultdict(
+        lambda: {'training': None, 'prediction': None, 'registration': None}
+    )
+    for alias_obj in civisml_template_alias_objects:
+        try:
+            job_type, version = _get_job_type_version(alias_obj.alias)
+        except ValueError:
+            msg = (
+                '%r looks like a CivisML alias for the prefix "civis-civisml-"'
+                ', but it is impossible to parse its job type and version'
+            )
+            log.debug(msg % alias_obj)
+            continue
```

Comment:
I've chosen to `log.debug` this if this unlikely situation arises. A warning would be visible to the user and it'd be at least confusing, if not incomprehensible, to them.